src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,148 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
```

Comment:
do you need this here?

Reply:
When I remove the "@SuppressWarnings("unchecked")" annotation, the "(T) objects[index]" warning appears. Which practice is better - with or without this annotation?

Reply:
This annotation should be put if you are sure that the compiler's warning can't happen. In this case, It would be better to use it at least not on the class level, but on the method level. For this task let's not use this annotation.